SYCL/ESIMD/api/functional/ctors/ctor_fill.hpp

```diff
@@ -0,0 +1,226 @@
+/// This file provides functions for tests on simd fill constructor.
+///
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "common.hpp"
+// For std::isnan
+#include <cmath>
+
+namespace esimd = sycl::ext::intel::experimental::esimd;
+namespace esimd_tests = esimd_test::api::functional::ctors;
+namespace esimd_functional = esimd_test::api::functional;
+
+// Descriptor class for the case of calling constructor in initializer context.
+struct initializer {
+  static std::string get_description() { return "initializer"; }
+
+  template <typename DataT, int NumElems>
+  static void call_simd_ctor(DataT init_value, DataT step, DataT *out) {
+    esimd::simd<DataT, NumElems> simd_by_init =
+        esimd::simd<DataT, NumElems>(init_value, step);
+    simd_by_init.copy_to(out);
+  }
+};
+
+// Descriptor class for the case of calling constructor in variable declaration
+// context.
+struct var_dec {
+  static std::string get_description() { return "variable declaration"; }
+
+  template <typename DataT, int NumElems>
+  static void call_simd_ctor(DataT init_value, DataT step, DataT *out) {
+    esimd::simd<DataT, NumElems> simd_by_var_decl(init_value, step);
+    simd_by_var_decl.copy_to(out);
+  }
+};
+
+// Descriptor class for the case of calling constructor in rvalue in an
+// expression context.
+struct rval_in_express {
+  static std::string get_description() { return "rvalue in an expression"; }
+
+  template <typename DataT, int NumElems>
+  static void call_simd_ctor(DataT init_value, DataT step, DataT *out) {
+    esimd::simd<DataT, NumElems> simd_by_rval;
+    simd_by_rval = esimd::simd<DataT, NumElems>(init_value, step);
+    simd_by_rval.copy_to(out);
+  }
+};
+
+// Descriptor class for the case of calling constructor in const reference
+// context.
+class const_ref {
+public:
+  static std::string get_description() { return "const reference"; }
+
+  template <typename DataT, int NumElems>
+  static void call_simd_ctor(DataT init_value, DataT step, DataT *out) {
+    return call_simd_by_const_ref<DataT, NumElems>(
+        esimd::simd<DataT, NumElems>(init_value, step), out);
+  }
+
+private:
+  template <typename DataT, int NumElems>
+  static void
+  call_simd_by_const_ref(const esimd::simd<DataT, NumElems> &simd_by_const_ref,
+                         DataT *out) {
+    simd_by_const_ref.copy_to(out);
+  }
+};
+
+// Enumeration of possible values for base value and step that will be provided
+// into simd constructor.
+enum class init_val {
+  min,
+  max,
+  zero,
+  min_half,
+  max_half,
+  neg_inf,
+  nan,
+  positive,
+  negative
+};
+
+// Dummy kernel for submitting some code into device side.
+template <typename DataT, int NumElems, typename T, init_val BaseVal,
+          init_val StepVal>
+struct kernel_for_fill;
+
+// Constructing a value for step and base values that depends on input
+// parameters.
+template <typename DataT, init_val BaseVal> DataT get_base_value() {
```

Comment:
Looks familiar a bit.. Why not just use some vector of reference data and iterate over it in runtime? Is it possible, or the code will bloat?

```suggestion
template <typename DataT, init_val BaseVal> DataT get_value() {
```
Nevertheless, looks like the `get_base_value` is used for both the `base` and `step` here)

Reply:
I've applied this comment [here](https://github.com/intel/llvm-test-suite/pull/598/commits/652dcd3b7c72dceecf81bb7bd486398d8664c24d#diff-7fc90b589428e6e89076f07f5ff280a217b7f5c8e836bfe6c742734825074ef4R104)